### PR TITLE
Change styles related to loop-flow

### LINKED ISSUE #565: Antares V8.2-rc1 - loop flows are highlighted in red in too many cases
**Description**
For a given timestamp, the loop flow value which is set by the user is highlighted in red in the GUI whenever for any transmission capacities time-series, the inequality -transmission capacity indirect <= loop flow <= transmission capacity direct is not verified.

**Expected behavior**
This highlight seems too restrictive as not all Time-Series will necessarily be used in the simulation. Having an orange highlight seems sufficient, as long as the inequality check mentioned in the description is clearly checked within the optimization step.

**OS and version**
 - OS: Windows
 - Antares Simulator version: 8.2-rc1

### PR DESCRIPTION
**Pull request by flomnes**
Fix #565 

* Change red (error) to orange (warning)
* Warning only if loop flow is enabled
* Cell style "warning" for transmission capacites if rule is violated

### CODE REVIEW

File: src/ui/simulator/toolbox/components/datagrid/renderer/connection.h [hunk 1]
```diff
@@ -92,6 +92,7 @@ class Connection : public Renderer::Matrix<>
 
 protected:
     wxWindow* pControl;
+    bool useLoopFlow = false;
```
Comment (guilpier-code) [inline src/ui/simulator/toolbox/components/datagrid/renderer/connection.h]: As **useLoopFlow** is a member variable, its name should reflect it (**useLoopFlow_** or **pUseLoopFlow**)

File: src/ui/simulator/toolbox/components/datagrid/renderer/connection.cpp [hunk 2]
```diff
@@ -137,26 +139,44 @@ bool connectionParameters::cellValue(int x, int y, const Yuni::String& value)
...
-        if (cellvalue < 0. && std::abs(cellvalue) > ntcIndirect)
+        if (loopflow < 0. && std::abs(loopflow) > ntcIndirect)
         {
             return false;
         }
     }
     return true;
 }
 
+static bool checkLoopFlowNTCDirect(double ntcDirect, double loopFlow, bool useLoopFlow)
```
Comment (guilpier-code) [inline src/ui/simulator/toolbox/components/datagrid/renderer/connection.cpp]: Function **checkLoopFlowNTCDirect** could be called inside function **checkLoopFlow**.
Same thing for **checkLoopFlowNTCIndirect**.

File: src/ui/simulator/toolbox/components/datagrid/renderer/connection.h [hunk 3]
```diff
@@ -137,9 +138,11 @@ class connectionNTCdirect : public connectionNTC
 public:
     connectionNTCdirect(wxWindow* parent, Toolbox::InputSelector::Connections* notifier);
     ~connectionNTCdirect() override = default;
+    IRenderer::CellStyle cellStyle(int col, int row) const override;
 
 private:
     void setMatrix(Data::AreaLink* link) override;
+    Antares::Matrix<>* parameters;
```
Comment (guilpier-code) [inline src/ui/simulator/toolbox/components/datagrid/renderer/connection.h]: You should make **parameters** a private variable in parent class **connectionNTC**.
You should change **parameters** into **parameters_** or **pParameters**

File: src/ui/simulator/toolbox/components/datagrid/renderer/connection.cpp [hunk 4]
```diff
@@ -265,6 +298,20 @@ connectionNTCindirect::connectionNTCindirect(wxWindow* parent,
...
+    parameters = link ? &(link->parameters) : nullptr;
+}
+
+IRenderer::CellStyle connectionNTCindirect::cellStyle(int col, int row) const
+{
+    if (!parameters)
+        return Renderer::Matrix<>::cellStyle(col, row);
+
+    const double loopFlow = (*parameters)[Data::fhlLoopFlow][row];
+    const double ntcInirect = Renderer::Matrix<>::cellNumericValue(col, row);
```
Comment (guilpier-code) [inline src/ui/simulator/toolbox/components/datagrid/renderer/connection.cpp]: **ntcInirect** should be changed into **ntcIndirect** (a **d** is missing)

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/ui/simulator/toolbox/components/datagrid/renderer/connection.h (hunk 1 above)
Comment (flomnes) [reply src/ui/simulator/toolbox/components/datagrid/renderer/connection.h]: done

File: src/ui/simulator/toolbox/components/datagrid/renderer/connection.cpp (hunk 2 above)
Comment (flomnes) [reply src/ui/simulator/toolbox/components/datagrid/renderer/connection.cpp]: done

File: src/ui/simulator/toolbox/components/datagrid/renderer/connection.cpp (hunk 4 above)
Comment (flomnes) [reply src/ui/simulator/toolbox/components/datagrid/renderer/connection.cpp]: done

File: src/ui/simulator/toolbox/components/datagrid/renderer/connection.h (hunk 3 above)
Comment (flomnes) [reply src/ui/simulator/toolbox/components/datagrid/renderer/connection.h]: You mean protected ? If private, child classes `connectionNTCindirect` and `connectionNTCdirect` can't access it, unless we be`friend` them.

**Comment by flomnes** [comment]
I applied all suggested changes. Notice also the small re-factor : I moved most of `cellStyle` one level up into `ConnectionNTC` and moved loopflow check functions (previously static) into `connectionNTCdirect` and `connectionNTCindirect`

File: src/ui/simulator/toolbox/components/datagrid/renderer/connection.cpp [hunk 5]
```diff
@@ -241,6 +247,19 @@ bool connectionNTC::cellValue(int x, int y, const Yuni::String& value)
...
 }
 
+IRenderer::CellStyle connectionNTC::cellStyle(int col, int row) const
+{
+    if (!mLoopFlowData)
+        return Renderer::Matrix<>::cellStyle(col, row);
+
+    const double loopFlow = (*mLoopFlowData)[row];
+    const double ntc = Renderer::Matrix<>::cellNumericValue(col, row);
+    if (!checkLoopFlow(ntc, loopFlow))
```
Comment (flomnes) [inline src/ui/simulator/toolbox/components/datagrid/renderer/connection.cpp:257]: pure virtual function, overriden in `connectionNTCdirect` and `connectionNTCindirect`

File: src/ui/simulator/toolbox/components/datagrid/renderer/connection.h (hunk 3 above)
Comment (guilpier-code) [reply src/ui/simulator/toolbox/components/datagrid/renderer/connection.h]: Yes, I mean **protected**, sorry. I meant private in the sense protected :-D

**Pull request merged**